koku/masu/processor/tasks.py

```diff
@@ -372,3 +373,79 @@ def vacuum_schema(schema_name):
                 cursor.execute(sql)
                 LOG.info(sql)
                 LOG.info(cursor.statusmessage)
+
+
+@app.task(name="masu.processor.tasks.autovacuum_tune_schema", queue_name="reporting")
+def autovacuum_tune_schema(schema_name):
+    """Set the autovacuum table settings based on table size for the specified schema."""
+    table_sql = """
+SELECT s.relname as "table_name",
+       s.n_live_tup,
+       coalesce(table_options.options, '{}'::jsonb) as "options"
+  FROM pg_stat_user_tables s
+  LEFT
+  JOIN (
+         select oid,
+                jsonb_object_agg(split_part(option, '=', 1), split_part(option, '=', 2)) as options
+           from (
+                  select oid,
+                         unnest(reloptions) as "option"
+                    from pg_class
+                   where reloptions is not null
+                ) table_options_raw
+          where option ~ '^autovacuum_vacuum_scale_factor'
+          group
+             by oid
+       ) as table_options
+    ON table_options.oid = s.relid
+ WHERE s.schemaname = %s
+ ORDER
+    BY s.n_live_tup desc;
+"""
+
+    alter_count = 0
+    xlrg_scale = Decimal("0.01")
+    lrg_scale = Decimal("0.02")
+    med_scale = Decimal("0.05")
+    no_scale = Decimal("100")
+    zero = Decimal("0")
+    reset = Decimal("-1")
+    scale_factor = zero
+    xlrg_table = 10000000
+    lrg_table = 1000000
+    med_table = 100000
```

Comment:
Does it make sense to expose any of these as tunable during runtime?  (e.g. as an environment variable) I think these are probably the knobs we'll want to fine-tune over time. So, exposing them would make that easier.

To keep it manageable, we could pass JSON as an environment variable, that way we can expose all of these settings with a single variable.

Reply:
That's a good idea. If we can modify the env at runtime then, sure.

Reply:
We can. If the data is exposed as an environment variable, then the DeploymentConfig sets the variable. So, we can change it at the DeploymentConfig level, which redeploys a new pod with the updated environment variables. It's a rolling update, so there's no downtime. But it means service operators can fine-tune the values without needing a new release.